cmd/execution_builder.go

```diff
@@ -208,7 +208,10 @@ func (builder *ExecutionNodeBuilder) LoadComponentsAndModules() {
 		Component("execution state", exeNode.LoadExecutionState).
 		Component("stop control", exeNode.LoadStopControl).
 		Component("execution state ledger WAL compactor", exeNode.LoadExecutionStateLedgerWALCompactor).
-		Component("execution data pruner", exeNode.LoadExecutionDataPruner).
+		// disable execution data pruner for now, since storehouse is going to need the execution data
+		// for recovery, will re-visit this once storehouse has implemented new WAL for checkpoint file of
+		// payloadless trie.
+		// Component("execution data pruner", exeNode.LoadExecutionDataPruner).
```

Comment:
ingestion engine is the only user of the execution data pruner, and [it is able to handle the nil case](https://github.com/onflow/flow-go/blob/leo/disable-execution-data-pruner/engine/execution/ingestion/engine.go#L462)